enhancements/declarative-index-config.md

```diff
@@ -88,35 +89,49 @@ $ cat etcd.json
         "base64data":"iVBORw0KGgoAAAANSUhEUgAAA.....",
         "mediatype":"image/png"
     },
-    "channels": ["alpha", "singlenamespace-alpha", "clusterwide-alpha"],
     "description": "A message about etcd operator, a description of channels"
-},
+}
 {
-    "schema": "olm.bundle", 
+    "schema": "olm.bundle",
     "name": "etcdoperator-community.v0.6.1",
     "package": "etcd",
     "image": "quay.io/operatorhubio/etcd:v0.6.1",
-    "version": "0.6.1",
     "properties":[
         {
-            "name": "olm.package",
-            "values": {
+            "type": "olm.package",
+            "value": {
+                "packageName": "etcd",
+                "version": "0.6.1"
+            }
+        },
+        {
+            "type": "olm.package.provided",
+            "value": {
                 "packageName": "etcd",
                 "version": "0.6.1"
             }
         },
         {
-            "name":"olm.gvk",
-            "type": "provided",
-            "values":{
-                "group": "etcd.database.coreos.com", 
-                "kind": "EtcdCluster", 
+            "type":"olm.gvk",
+            "value": {
+                "group": "etcd.database.coreos.com",
+                "kind": "EtcdCluster",
+                "version": "v1beta2"
+            }
+        },
+        {
+            "type":"olm.gvk.provided",
+            "value": {
+                "group": "etcd.database.coreos.com",
+                "kind": "EtcdCluster",
                 "version": "v1beta2"
             }
         },
```

Comment:
See the note in the "Note on properties" section about why this duplication is present.

Reply:
(this doesn't have to block this PR) but @bparees we could retain the `channels` attribute for `olm.package` with a similar validation protocol. (this is based on the assumption that we're fine with calculating the value of `channels` every time we need it and that's the direction we've landed on in this PR)